cloudevents/encoding_binary.go

```diff
@@ -76,14 +77,21 @@ type BinaryLoader interface {
 }
 
 // FromRequest parses event data and context from an HTTP request.
-func (binary) FromRequest(data interface{}, r *http.Request) (*EventContext, error) {
-	// TODO: detect version and create correct VXXEventContext
-	ec := &EventContext{}
+func (binary) FromRequest(data interface{}, r *http.Request) (LoadContext, error) {
+	var ec LoadContext
+	if r.Header.Get("CE-SpecVersion") == V02CloudEventsVersion {
```

Comment:
I really prefer `switch` in this case, especially i v03 is coming at some point.
```
switch {
 case r.Header....:
...
default:
  error
}
```

Reply:
That would be easier if these two used the same field on the map. (Yes, they changed their version sentinel between v0.1 and v0.2...)

I can change this one over anyway.